From 26e95c876831692f28894041c5a1db5934715939 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.org>
Date: Wed, 18 Dec 2019 10:41:33 +0000
Subject: [PATCH] overlays: dht11: Allow multiple instantiation

Add addresses to the dht11 and dht11_pins nodes to allow unique names
to be generated by assigning to the "reg" property.

Signed-off-by: Phil Elwell <phil@raspberrypi.org>
---
 arch/arm/boot/dts/overlays/dht11-overlay.dts | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

--- a/arch/arm/boot/dts/overlays/dht11-overlay.dts
+++ b/arch/arm/boot/dts/overlays/dht11-overlay.dts
@@ -24,7 +24,7 @@
 	fragment@1 {
 		target = <&gpio>;
 		__overlay__ {
-			dht11_pins: dht11_pins {
+			dht11_pins: dht11_pins@0 {
 				brcm,pins = <4>;
 				brcm,function = <0>; // in
 				brcm,pull = <0>; // off
@@ -34,6 +34,8 @@
 
 	__overrides__ {
 		gpiopin = <&dht11_pins>,"brcm,pins:0",
-			<&dht11>,"gpios:4";
+			<&dht11_pins>, "reg:0",
+			<&dht11>,"gpios:4",
+			<&dht11>,"reg:0";
 	};
 };
